scripts/calculateRewards/proofOfImpact.ts

```diff
@@ -0,0 +1,122 @@
+import yargs from 'yargs'
+import { parse } from 'csv-parse/sync'
+import { readFileSync } from 'fs'
+import { parseEther } from 'viem'
+import BigNumber from 'bignumber.js'
+
+// proof-of-impact campaign parameters
+// May 8 2025 12:00:00 AM UTC
+const proofOfImpactStartTimestamp = '1746687600000'
+// May 29 2025 12:00:00 AM UTC
+const proofOfImpactEndTimestamp = '1748502000000'
+const totalRewards = parseEther('14839')
+
+const rewardsPerMillisecond = new BigNumber(totalRewards).div(
+  new BigNumber(proofOfImpactEndTimestamp).minus(
+    new BigNumber(proofOfImpactStartTimestamp),
+  ),
+)
+
+export const _rewardsPerMillisecond = rewardsPerMillisecond // for testing
+
+export function calculateRewardsProofOfImpact({
+  kpiData,
+  startTimestamp,
+  endTimestamp,
+}: {
+  kpiData: KpiRow[]
+  startTimestamp: string
+  endTimestamp: string
+}) {
+  const timeDiff = new BigNumber(endTimestamp).minus(
+    new BigNumber(startTimestamp),
+  )
+  const totalRewardsForPeriod = timeDiff.times(rewardsPerMillisecond)
+
+  const referrerKpis = kpiData.reduce(
+    (acc, row) => {
+      if (!(row.referrerId in acc)) {
+        acc[row.referrerId] = BigInt(row.revenue)
+      } else {
+        acc[row.referrerId] += BigInt(row.revenue)
+      }
+      return acc
+    },
+    {} as Record<string, bigint>,
+  )
+
+  const total = Object.values(referrerKpis).reduce(
+    (sum, value) => sum + value,
+    BigInt(0),
+  )
+
+  const rewards = Object.entries(referrerKpis).map(([referrerId, kpi]) => {
+    return {
+      referrerId,
+      kpi,
+      rewardAmount: totalRewardsForPeriod.times(kpi).div(total).toString(),
+    }
+  })
+
+  return rewards
+}
+
+function parseArgs() {
+  return yargs
+    .option('input-file', {
+      alias: 'i',
+      description: 'input file path containing revenue data',
+      type: 'string',
+      demandOption: false,
+    })
+    .option('output-file', {
+      alias: 'o',
+      description: 'output file path to write reward allocations',
+      type: 'string',
+      demandOption: false,
+    })
+    .option('start-timestamp', {
+      alias: 's',
+      description: 'start timestamp',
+      type: 'string',
+      demandOption: true,
+    })
+    .option('end-timestamp', {
+      alias: 'e',
+      description: 'end timestamp',
+      type: 'string',
+      demandOption: true,
+    })
+    .strict()
+    .parseSync()
+}
+
+interface KpiRow {
+  referrerId: string
+  userAddress: string
+  revenue: string
+}
+
+async function main(args: ReturnType<typeof parseArgs>) {
+  const inputPath = args['input-file'] ?? `${args['protocol']}-revenue.csv`
```

Comment:
This `protocol` argument doesn't look to be defined?

Reply:
oh yeah, lol 😅 i added a hard coded fallback!